Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml

```diff
@@ -0,0 +1,131 @@
+      2, "High", 
+      3, "Medium",
+      4, "Low",
+      5, "Low", 
+      6, "Informational",
+      7, "Informational" 
+  ];
+  let EventResultWFLookup = datatable (
+      Action_s: string,
+      EventResult_WF: string,
+      DvcAction: string
+  )
+      [
+      "LOG", "Success", "ALLOW",
+      "DENY", "Failure", "DENY",
+      "WARNING", "Success", "WARNING"
+  ];
+  let EventTypeLookup = datatable (LogType_s: string, EventType: string, EventOriginalType: string )
+      [
+      "WF", "HTTPsession", "Web Firewall",
+      "TR", "WebServerSession", "Access"
+  ];
+  let parser=(disabled: bool=false) {
+      barracuda_CL
+      | where not(disabled) and (LogType_s == "WF" or LogType_s == "TR")
+      | lookup EventResultWFLookup on Action_s
+      | lookup EventTypeLookup on LogType_s
+      | extend
+          severity = toint(Severity_s)
+      | lookup SeverityLookup on severity
+      | extend
+          Dst = column_ifexists("ServerIP_s", ""),
+          EventCount = toint(1),
+          EventProduct = "WAF",
+          EventSchema = "WebSession",
+          EventSchemaVersion = "0.2.6",
+          EventVendor = "Barracuda",
+          status_code = toint(column_ifexists("HTTPStatus_s", ""))
+      | extend EventResult_TR = case(
+                                status_code between (200 .. 299),
+                                "Success", 
+                                status_code between (400 .. 599),
+                                "Failure",
+                                status_code between (300 .. 399),
+                                "Partial",
+                                "NA"
+                            )
+      | extend
+          Dvc = column_ifexists("UnitName_s", ""),
+          EventResult = iff(LogType_s == "TR", EventResult_TR, EventResult_WF),
+          EventStartTime = unixtime_milliseconds_todatetime(tolong(DeviceReceiptTime_s)),
+          DstIpAddr = column_ifexists("ServerIP_s", ""),
+          SrcIpAddr = column_ifexists("ClientIP_s", ""),
+          DstBytes = tolong(column_ifexists("BytesReceived_d", "")),
+          DstPortNumber = toint(column_ifexists("ServerPort_d", "")),
+          HttpCookie = column_ifexists("Cookie_s", ""),
+          HttpReferrer = column_ifexists("Referer_s", ""),
+          HttpRequestBodyBytes = tolong(column_ifexists("BytesReceived_d", "")),
+          HttpRequestMethod = column_ifexists("Method_s", ""),
+          HttpResponseBodyBytes = tolong(column_ifexists("BytesSent_d", "")),
+          HttpResponseTime = toint(column_ifexists("TimeTaken_d", "")),
+          HttpUserAgent = column_ifexists("UserAgent_s", ""),
+          NetworkSessionId = column_ifexists("SessionID_s", ""),
+          Rule = column_ifexists("RuleType_s", ""),
+          SrcPortNumber = toint(column_ifexists("ClientPort_d", "")),
+          SrcUsername = column_ifexists("AuthenticatedUser_s", ""),
+          Url = column_ifexists("URL_s", ""),
+          HttpResponseCacheControl = iff(
```

Comment:
I don't see this field in the schema or tester. Where have we taken this field from?

Reply:
We have taken this from the [ASimTester.csv](https://github.com/Azure/Azure-Sentinel/blob/users/demehra/ASIMTestAudit/ASIM/dev/ASimTester/ASimTester.csv) file.
<img width="743" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/2c3c6692-0853-4906-b007-d47839992634">